detect.go

```diff
@@ -29,24 +30,59 @@ type ProjectParser interface {
 	NodeIsRequired(path string) (bool, error)
 }
 
-func Detect(buildpackYMLParser BuildpackConfigParser, configParser ConfigParser, projectParser ProjectParser) packit.DetectFunc {
+// Detect will return a packit.DetectFunc that will be invoked during the
+// detect phase of the buildpack lifecycle.
+//
+// Detection will contribute a Build Plan that requires different things
+// depending on the type of app being built. See Configuration for details
+// on how environment variable configuration influences detection.
+//
+// Source Code Apps
+//
+// The buildpack will require the .NET Core SDK at build-time and the .NET Core
+// Runtime and ASP.NET Core at launch-time. It will require ICU at launch time.
+// It will require Nodejs at launch time if the app relies on JavaScript
+// components.
+//
+// Framework-dependent Deployments
+//
+// The buildpack will require the .NET Core Runtime and ASP.NET Core at
+// launch-time to run the framework-dependent app. It will require ICU at
+// launch time. It will require the .NET Core SDK at launch-time so that the
+// dotnet CLI is available to invoke the app's entrypoint DLL. It will require
+// Nodejs if the app relies on JavaScript components.
+//
+// Framework-dependent Executables
+//
+// The buildpack will require the .NET Core Runtime and ASP.NET Core at
+// launch-time to run the framework-dependent app. It will require ICU at
+// launch time. It will require Nodejs at launch time if the app relies on JavaScript
+// components.
+//
+// Self-contained Executables
+// The buildpack will require ICU at launch time. It will require Nodejs at
+// launch time if the app relies on JavaScript components.
+func Detect(
+	config Configuration,
+	logger scribe.Emitter,
+	buildpackYMLParser BuildpackConfigParser,
+	configParser ConfigParser,
+	projectParser ProjectParser,
+) packit.DetectFunc {
 	return func(context packit.DetectContext) (packit.DetectResult, error) {
-		var projectPath string
-		var ok bool
-		var err error
-
-		if projectPath, ok = os.LookupEnv("BP_DOTNET_PROJECT_PATH"); !ok {
-			projectPath, err = buildpackYMLParser.ParseProjectPath(filepath.Join(context.WorkingDir, "buildpack.yml"))
-			if err != nil {
-				return packit.DetectResult{}, fmt.Errorf("failed to parse buildpack.yml: %w", err)
-			}
+		logger.Debug.Process("Build configuration:")
+		es, err := env.Marshal(&config)
```

Comment:
Is there a reason for not using the new `logger.BuildConfiguration` function?

Reply:
Because of that bug which will end up printing the `default` tag. I figure for now it's just easier to do my own logging here.